### ISSUE TITLE
Saving a PNG as Jpeg only processes a part of the image on .NET 6

### ISSUE THREAD

**Issue by kerams** [SixLabors/ImageSharp#1704]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Loading a PNG file and immediately saving it as Jpeg works as expected on .NET 5. However when I switch to `net6.0` in the csproj, only the top part of the image appears in the output jpeg file, the rest is black. The dimensions are preserved. Feel free to close this if you believe it's a bug in .NET and not anything to do with ImageSharp.

### Steps to Reproduce
```csharp
using var img = Image.Load("unknown.png");
var fs = File.OpenWrite("unknown.jpeg");
img.SaveAsJpeg(fs);
fs.Flush();
fs.Close();
```

Use this image
https://user-images.githubusercontent.com/5063478/126033585-4efb31ce-c231-4229-8412-68fb73950ffc.png

The result ont .NET 6 Preview 6 looks like this
https://user-images.githubusercontent.com/5063478/126033623-0f886133-bc0d-4449-8048-d5d29a0e377c.jpeg

### System Configuration
- ImageSharp version: 1.0.3
- Environment (Operating system, version and so on): Win 10
- .NET Framework version:
  - 5.0.400-preview.21277.10
  - 6.0.100-preview.6.21355.2

**Comment by JimBobSquarePants** [comment]
You should raise this issue in the .NET runtime.

**Comment by kerams** [comment]
I could do that, but I'm not sure how they're going to investigate what from their point of view would appear to be a corrupted image generation in a third party library.

**Comment by JimBobSquarePants** [comment]
It’ll be treated as a regression since it works in all previous frameworks. Our code hasn’t changed. 

We had similar issues with the .NET 5 previews where the runtime was emitting incorrect values for some intrinsics instructions.

**Comment by br3aker** [comment]
Image is corrupted in a very funny way:
1. Jpeg header
2. Pixels (visible rows at the top)
3. A lot of black pixels (visible bottom rows)
4. EOI terminator (as expected)
5. **Actual color data which should have been written in place of black pixels** ending with second EOI marker

Simply removing that black mess from binary reveals that encoder actually encoded color values:
https://user-images.githubusercontent.com/20967409/126034657-66232514-f099-4dff-a052-2cc44e5555ad.jpeg

**TLDR;**
have no idea why I'm writing this but yep, there's no way encoder would put EOI marker not only twice but in the middle of color data. There's definately something strange happening within 6.0.

**Comment by brianpopow** [comment]
Out of curiosity i have run our unit tests with .net 6.0. There are 70 failing tests. Mainly the png decoder is affected (seems similar to the jpeg issue). Also some WuQuantizer tests and TiffDecoder tests fail.

Im using 6.0.100-preview.6.21355.2
